packages/mdc-floating-label/README.md

```diff
@@ -75,33 +75,32 @@ CSS Class | Description
 
 Mixin | Description
 --- | ---
-`mdc-floating-label-color($color)` | Customizes the color of the label.
+`mdc-floating-label-ink-color($color)` | Customizes the ink color of the label.
 `mdc-floating-label-fill-color($color)` | Customizes the fill color of the label.
-`mdc-floating-label-invalid-shake-keyframes($modifier, $positionY, $positionX, $scale)` | Creates a shake keyframe animation for invalid label shake. To be used with sass mixin `mdc-floating-label-shake($modifier)`.
-`mdc-floating-label-shake($modifier)` | Applies shake keyframe animation to label.
+`mdc-floating-label-shake-keyframes($modifier, $positionY, $positionX, $scale)` | Generates a CSS `@keyframes` at-rule for an invalid label shake. Used in conjunction with the `mdc-floating-label-shake-animation` mixin.
+`mdc-floating-label-shake-animation($modifier)` | Applies shake keyframe animation to label.
 `mdc-floating-label-float-position($positionY, $positionX, $scale)` | Sets position of label when floating.
 
 ### `MDCFloatingLabel`
 
 Method Signature | Description
 --- | ---
-`shake(isValid: boolean, isFocused: boolean) => void` | Shakes label when invalid
-`float(value: string, isFocused: boolean, isBadInput: boolean) => void` | Floats label above text field or select
-`getWidth() => number` | Returns the width of the label element
-
+`shake(isValid: boolean, isFocused: boolean) => void` | Label animation class is applied unless isValid or isFocused is true.
```

Comment:
I think we should just have a single boolean param called `shouldShake`, and let the caller determine its value.

What you have now is a "leaky abstraction": text-field logic is leaking into a different component.